vms/avm/block/builder/builder.go

```diff
@@ -93,8 +93,8 @@ func (b *builder) BuildBlock(context.Context) (snowman.Block, error) {
 		remainingSize = targetBlockSize
 	)
 	for {
-		tx := b.mempool.Peek(remainingSize)
-		if tx == nil {
+		tx := b.mempool.Peek()
+		if tx == nil || len(tx.Bytes()) > remainingSize {
 			break
```

Comment:
You may wish to stop only if you are at least some minimum target block pack. It is possible you have something like: `[1,7,1,1,1,1]`.

Reply:
In other words...if you can't put a tx in but the block is only 10% full, keep trying other transactions.